py/objdict.c

```diff
@@ -143,6 +143,21 @@ STATIC mp_obj_t dict_unary_op(mp_unary_op_t op, mp_obj_t self_in) {
     }
 }
 
+STATIC void dict_union_helper(mp_obj_dict_t *self, mp_obj_t other) {
```

Comment:
I have not read thoroughly through everything, but could this function helper be unnecessary. I see that the proposal mentiones dict update, https://www.python.org/dev/peps/pep-0584/ 
Could that one be used instead?